### PR TITLE
Check that Webhook events we receive belong to correct repository

### PR DESCRIPTION
**Pull request by mrnugget**
Another approach to https://github.com/sourcegraph/sourcegraph/pull/8997

### CODE REVIEW

File: enterprise/internal/campaigns/webhooks.go [hunk 1]
```diff
@@ -287,7 +306,7 @@ func (h *GitHubWebhook) convertEvent(ctx context.Context, theirs interface{}) (p
 		for _, pr := range cr.PullRequests {
 			n := pr.GetNumber()
 			if n != 0 {
-				prs = append(prs, int64(n))
+				prs = append(prs, PR{ID: int64(n)})
```
Comment (eseliger) [inline enterprise/internal/campaigns/webhooks.go]: `RepoID` is missing :)

Comment (mrnugget) [reply enterprise/internal/campaigns/webhooks.go]: The  👁 Eye of Erik 👁 strikes again

File: enterprise/internal/campaigns/webhooks.go [hunk 2]
```diff
@@ -25,12 +29,20 @@ type Webhook struct {
 	Repos repos.Store
 	Now   func() time.Time
 
-	Service string
+	// ServiceType corresponds to api.ExternalRepoSpec.ServiceType
+	// Example values: bitbucketserver.ServiceTyp, github.ServiceType
```
Comment (mrnugget) [inline enterprise/internal/campaigns/webhooks.go]:
```suggestion
	// Example values: bitbucketserver.ServiceType, github.ServiceType
```

**Comment by mrnugget** [comment]
@ryanslade I pair-reviewed this with @eseliger. Merging this now. I can address comments in a follow-up PR.

File: cmd/repo-updater/repos/bitbucketserver_test.go [hunk 3]
```diff
@@ -270,7 +270,7 @@ func TestBitbucketServerSource_CreateChangeset(t *testing.T) {
 				Title:     "This is a test PR",
 				Body:      "This is the body of a test PR",
 				BaseRef:   "master",
-				HeadRef:   "test-pr-bbs-9",
+				HeadRef:   "test-pr-bbs-11",
```
[the remark this answers is not in the document]
Comment (mrnugget) [reply cmd/repo-updater/repos/bitbucketserver_test.go]: The test is for creating a new pull request and making sure that it worked — that only works once. That's why, when updating the test data, you need to create a new branch and new PR.

**Comment by mrnugget** [comment]
@unknwon I'll address your feedback on Monday!

**Pull request merged**